pymc/model.py

```diff
@@ -1455,14 +1457,18 @@ def add_random_variable(self, var, dims: Optional[Tuple[Union[str, None], ...]]
             setattr(self, self.name_of(var.name), var)
 
     @property
-    def prefix(self):
-        return f"{self.name}_" if self.name else ""
+    def prefix(self) -> str:
+        if self.isroot or not self.parent.prefix:
+            name = self.name
+        else:
+            name = f"{self.parent.prefix}/{self.name}"
+        return name.strip("/")
```

Comment:
Should we raise if a user tries to create a variable with `/` in the name?

Reply:
I think it should not matter since model name is scope and `scope/` seems to be intuitively valid

Reply:
Thinking about it again I don't think there's any issue here. I was thinking about variable names starting/ending with `/`, but this is just for the prefix part